chalice/app.py

```diff
@@ -820,6 +922,22 @@ def __call__(self, event, context):
         response = response.to_dict(self.api.binary_types)
         return response
 
+    def __call__(self, event, context):
+        # This is what's invoked via lambda.
+        # Sometimes the event can be something that's not
+        # what we specified in our request_template mapping.
+        # When that happens, we want to give a better error message here.
+        route_key = event.get('requestContext', {}).get('routeKey')
+        resource_path = event.get('requestContext', {}).get('resourcePath')
+        if route_key is not None:
+            return self._handle_websocket(route_key, event, context)
```

Comment:
I'd prefer we not reuse the same lambda function for both rest APIs and websocket APIs.  We've had feedback that customers would even like the individual route handlers to be separate lambda functions.

I think at a minimum we should have a separate lambda function for websockets, and a separate one for rest APIs.  I wouldn't be surprised if we get asked for separate functions for connect/disconnect as well.

Reply:
there's several places this notion is wired in swagger generation that needs to be addressed to resolve that end goal of multiple functions on routes, most of which is fixing the calling abstractions to pass in better context, atm they reference a static string ref due to the separation of context and need to reference a common var. is there an extant issue re that function route separation?

Reply:
Just following up here, it'd be nice to have the websocket handler code as a separate class, similar to what we've done for new event handlers.  That way they're decoupled from the `Chalice` object.